ArmEmulator/Main.fs

```diff
@@ -2,10 +2,144 @@ module ArmEmulator
 
 open Expecto
 open VisualTest
+open System.IO
+open TopLevel
+open CommonData
 
-[<EntryPoint>]
-let main argv =
+/// reads the lines of a file into a string
+let readLines (filePath:string) = seq {
+    use sr = new StreamReader (filePath)
+    while not sr.EndOfStream do
+        yield sr.ReadLine ()
+}
+
+/// converts a DataPath object to valid JSON as a string
+let cpuToJson cpu = 
+    let regsJson = 
+        cpu.Regs 
+        |> Map.toList
+        |> List.map snd
+        |> fun x -> (sprintf "%A" x).Replace("u","").Replace(";",",")
+        |> sprintf "'regs': %s"
+    let flagsJson = 
+        (cpu.Fl.N, cpu.Fl.Z, cpu.Fl.C, cpu.Fl.V)
+        |> fun (n,z,c,v)-> 
+            sprintf "{'N': %A, 'Z': %A, 'C': %A, 'V':%A}" n z c v
+        |> sprintf "'flags': %s"
+    let memJson = 
+        match Map.toList cpu.MM with
+        | [] -> "'mem': {}"
+        | lst -> 
+            lst
+            |> List.map (
+                fun (WA a, (DataLoc v | Code v)) -> sprintf "'%A': %A, " a v)
+            |> List.reduce (+)
+            |> fun x -> x.Replace("u","")
+            |> sprintf "'mem': {%s}"
+            |> fun x -> x.Replace(", }","}")
+    (regsJson, flagsJson, memJson)
+    |||> sprintf "{%s, %s, %s}" 
+
+/// runs the emulator based on a source file in filePath
```

Comment:
So is the idea here to save the contents of the monaco editor to a file before execution?

Reply:
No this is separate to the monaco editor stuff. Basically it just lets you run the emulator from the command line. All of the monaco interfacing will happen in the `TopLevel` module.